chain_of_responsibility/client.go

```diff
@@ -0,0 +1,14 @@
+package chain_of_responsibility
+
+// ProcessCustomerRequest : externally exposed
+// method for top layer to trigger complain
+// processing flows based on some paramters
+func ProcessCustomerRequest(complainType string) string {
```

Comment:
better way here is to create list of the baseHandler, iterate the list from 0th index to n-1th index, and call the setNext function for each node. 

var handlers []baseHandler = {newHandler1(),newHandler2(),newHandler3()}
for i:=0;i<len(handlers)-1;i++{
handlers[i].setNext(handlers[i+1])
}